### PR TITLE
RTN6

### PR DESCRIPTION
**Pull request by ricardopereira**

### CODE REVIEW

**Comment by ricardopereira** [comment]
I needed cdebf4b, 00ae3c0, ccccf2a, 225dbd8 from branch `RSA15a`.

File: ably-ios/ARTConnectionDetails.m [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    }
+    return self;
+}
+
+- (NSString *)getClientId {
+    return _protocolMessage.clientId;
+}
+
+- (NSString *)getConnectionKey {
+    return _protocolMessage.connectionKey;
```
Comment (mattheworiordan) [inline ably-ios/ARTConnectionDetails.m]: FYI, we prefer `ConnectionDetails#connectionKey` over `ProtocolMessage#connectionKey`, see http://docs.ably.io/client-lib-development-guide/features/#CD2b

File: ablySpec/TestUtilities.swift [hunk 2]
```diff
@@ -327,6 +327,8 @@ class MockHTTPExecutor: NSObject, ARTHTTPExecutor {
 class MockTransport: ARTWebSocketTransport {
 
     var lastUrl: NSURL?
+    var lastSentMessage: ARTProtocolMessage?
+    var connectedMessage: ARTProtocolMessage?
```
Comment (mattheworiordan) [inline ablySpec/TestUtilities.swift]: BTW. MockTransport is not a great name given it's not really a mock, but instead a Proxy.

**Comment by mattheworiordan** [comment]
Looks good, would be nice to consider better naming for `MockTransport` perhaps?

File: ably-ios/ARTConnectionDetails.m (hunk 1 above)
Comment (ricardopereira) [inline ably-ios/ARTConnectionDetails.m]: @mattheworiordan So the `connectionKey` is stored when the RealtimeClient receives a CONNECTED message and that is it?

**Comment by ricardopereira** [comment]
@mattheworiordan Yes, like `TestProxyTransport` and maybe changing for `MockHTTPExecutor` as well.

**Comment by mattheworiordan** [comment]
> Yes, like TestProxyTransport and maybe changing for MockHTTPExecutor as well.

Sounds good

File: ably-ios/ARTConnectionDetails.m (hunk 1 above)
Comment (mattheworiordan) [inline ably-ios/ARTConnectionDetails.m]: Yes, connection Key is only stored once for every CONNECTED message received, which should only ever be received when you open, recover or resume a connection.

Comment (ricardopereira) [inline ably-ios/ARTConnectionDetails.m]: Opened issue #118

**Comment by ricardopereira** [comment]
@mattheworiordan Done

**Comment by mattheworiordan** [comment]
Thanks

**Pull request merged**